miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/ListUsersController.java

```diff
@@ -24,78 +24,57 @@
 package uk.ac.bbsrc.tgac.miso.webapp.controller;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.core.session.SessionInformation;
-import org.springframework.security.core.session.SessionRegistryImpl;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.servlet.ModelAndView;
 
+import com.eaglegenomics.simlims.core.User;
 import com.eaglegenomics.simlims.core.manager.SecurityManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.webapp.util.ListItemsPage;
 
 @Controller
 public class ListUsersController {
-  protected static final Logger log = LoggerFactory.getLogger(ListUsersController.class);
-
   @Autowired
   private SecurityManager securityManager;
 
-  @Autowired
-  private SessionRegistryImpl sessionRegistry;
+  private final ListItemsPage usersPage = new ListItemsPage("user") {
 
-  public void setSecurityManager(SecurityManager securityManager) {
-    assert (securityManager != null);
-    this.securityManager = securityManager;
-  }
+    @Override
+    protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) throws IOException {
+      User user = securityManager.getUserByLoginName(SecurityContextHolder.getContext().getAuthentication().getName());
+      config.put("isAdmin", user.isAdmin());
+      config.put("isTech", Arrays.asList(user.getRoles()).contains("ROLE_TECH"));
+    }
 
-  @ModelAttribute("title")
-  public String title() {
-    return "Users";
-  }
+  };
 
   @RequestMapping("/admin/users")
   public ModelAndView adminListUsers(ModelMap model) throws IOException {
-    try {
-
-      log.info("Logged in users: " + sessionRegistry.getAllPrincipals().size());
-
-      List<Object> nonExpiredUsers = new ArrayList<Object>();
-      for (Object o : sessionRegistry.getAllPrincipals()) {
-        List<SessionInformation> sessinfo = sessionRegistry.getAllSessions(o, false);
-        for (SessionInformation si : sessinfo) {
-          if (!si.isExpired()) {
-            nonExpiredUsers.add(o);
-          }
-        }
-      }
-      model.addAttribute("loggedInUsers", nonExpiredUsers);
```

Comment:
This appears to be a pre-existing issue instead of a newly-introduced one. I'm going to approve this on the basis of pre-existence, and if you're checking for logged-in users then you can hopefully figure out that you are logged in.